icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6665,6 +6667,145 @@ static void test_unicode_define(const char *namech, char ch, const char *nameu,
   }
 }
 
+static void checkTerminating(
+    const char* locale, const char* inLocale) {
+  UErrorCode status = U_ZERO_ERROR;
+  int32_t preflight_length = uloc_getDisplayName(
+      locale, inLocale, NULL, 0, &status);
+  if (status != U_BUFFER_OVERFLOW_ERROR) {
+      log_err("uloc_getDisplayName(%s, %s) preflight failed", locale, inLocale);
```

Comment:
If the test fails here, I wonder if it should just `return` right away?
(Same with the other `log_err` below)

Reply:
Makes sense, but I would accept it either way.